tests/integration-tests/tests/update/test_update/test_dynamic_file_systems_update/storage-stack.yaml

```diff
@@ -211,68 +359,219 @@ Resources:
+                  DataRepositoryAssociations=[
+                    {
+                      "FileCachePath": file_cache_path,
+                      "DataRepositoryPath": s3_bucket,
+                    },
+                  ],
+                ).get("FileCache")
+              )
 
-# Using the tests/integration-tests/resources/file-cache-storage-cfn.yaml which is uploaded in S3 bucket
-  FileCacheStack:
-    Condition: CreateFsx
-    Type: AWS::CloudFormation::Stack
+            except fsx.exceptions.InternalServerError as e:
+              logger.error(f"Cannot create FileCache because of InternalServerError.")
+              raise Exception(f"Cannot create FileCache {e}")
+
+
+          def handler(event, context):
+            logger.info(f"Context: {context}")
+            logger.info(f"Event: {event}")
+            logger.info(f"Boto version: {boto3.__version__}")
+
+
+
+            response_data = {}
+            reason = None
+            response_status = cfnresponse.SUCCESS
+
+            try:
+              physical_resource_id = ""
+              if event['RequestType'] == 'Create':
+                file_cache_path =  event['ResourceProperties']['FileCachePath']
+                logger.info(f"file_cache_path: {file_cache_path}")
+
+                s3_bucket = 's3://{}'.format(event['ResourceProperties']['S3BucketName'])
+                logger.info(f"s3_bucket: {s3_bucket}")
+
+                subnet_id = event['ResourceProperties']['SubnetId']
+                logger.info(f"subnet_id: {subnet_id}")
+
+                securitygroup_id = event['ResourceProperties']['SecurityGroupId']
+                logger.info(f"securitygroup_id: {securitygroup_id}")
+                file_cache_response = create_file_cache_id(file_cache_path, s3_bucket, subnet_id, securitygroup_id)
+                file_cache_id = file_cache_response.get("FileCacheId")
+                file_cache_arn= file_cache_response.get("ResourceARN")
+                response_data['Message'] = 'Resource creation successful!'
+                physical_resource_id = file_cache_id
+
+                logger.info(f"file_cache_id: {file_cache_id}")
+                # provide outputs
+                response_data['FileCacheId'] = file_cache_id
+                response_data['FileCacheARN'] = file_cache_arn
+              elif event['RequestType'] == 'Delete':
+                physical_resource_id = event.get("PhysicalResourceId")
+                logger.info('file_cache_id {}'.format(physical_resource_id))
+                delete_file_cache_id(file_cache_id=physical_resource_id)
+
+              else:
+                  physical_resource_id = event['PhysicalResourceId']
+            except Exception as e:
+                response_status = cfnresponse.FAILED
+                logger.error(f"Exception {e}")
+                reason = str(e)
+            cfnresponse.send(event, context, response_status, response_data, physical_resource_id, reason)
+
+  CreateDeleteFileCache:
```

Comment:
Do we need the `file-cache-storage-cfn.yaml` if we have moved it here?

Reply:
Currently, yes we do as it is used by another integ test `test_file_cache`.
In another iteration we will make every integ test requiring external storage use the new storage stack and select the storage it is interested in from parameters.